odk2-src/build-app.rst

```diff
@@ -0,0 +1,795 @@
+
+There are several times during app development where you will need to push and pull files to and from the phone. You will have to open one of the ODK tools on the device before these commands succeed.
+
+ - The :command:`push` command is used to push the entire app directory to the mobile device.
+ - The :command:`pull` command is used to pull the database or exported CSVs from the device to the desktop computer.
+
+.. tip::
+  Exported CSVs can be used to set up :file:`tables.init` to load test data.
+
+:program:`Grunt` tasks have been written in :file:`Gruntfile.js` that perform these operations for you.
+
+These commands can be run anywhere within the :file:`Application Designer` directory.
+
+  - :command:`grunt adbpush`: Pushes everything under the app directory to the device.
+  - :command:`grunt adbpull-db`: Pulls the database from the device to the PC.
+  - :command:`grunt adbpull-csv`: Pull the exported CSVs from the device to the PC.
+
+The pull commands will place the pulled content in the :file:`app/output/` directory.
+
+The database is a :program:`SQLite` database and can be viewed using :program:`SQLite Browser`. This tool can also be used to view the content of the database used by :program:`Chrome` on your computer (the location of that file is OS dependent).
+
+If you pull the CSV files, they will be under the :file:`output/csv/` directory. You can then copy them to the :file:`config/assets/csv/` directory and set up the :file:`tables.init` file to read them in order to provision test data for your development effort. If you need any of this data in production, you will want to sync to a server then export the CSV files and copy them to the :file:`config/assets/csv/` directory so that they have all of their metadata field values populated.
+
+.. tip::
+  Running :command:`grunt adbpull` will perform all the pull tasks.
+
+.. tip::
+  There are a number of additional grunt tasks available. Assuming you have installed grunt and node, you can view the available tasks by running :command:`grunt --help` anywhere in the repo.
+
+
+.. _build-app-deploying:
+
+Deploying an Application
+----------------------------
+
+This step requires that you first set up a :doc:`cloud-endpoints-intro`.
+
+  1. Push your application to a clean device (guide: :ref:`build-app-pushing`).
+
+  2. Authenticate as a user in the table administrator group (guide: :ref:`services-using-change-user`).
+
+  3. Reset the App Server (guide: :ref:`services-using-reset-app-server`).
+
+The application is now deployed to your server. Other devices can synchronize with that server to download the application and start collected data.
+
+.. _build-app-deploying-updating:
+
+Updating an Application
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To update any app level or table level files, or to modify the database schema, you will need to reset the app server. Make the changes on your PC as normal, push them to the device, and reset the app server.
```

Comment:
```patch
@@ -769,9 +788,7 @@ The application is now deployed to your server. Other devices can synchronize wi
 Updating an Application
 ~~~~~~~~~~~~~~~~~~~~~~~~~
 
-  - **Changing view files**: In this scenario you are not changing the database schema and your changes only affect the presentation. These changes can be synchronized to the server and do not require resetting the app server.
-
-  - **Changing the database schema**: In this scenario you want to change the database schema, such as adding a new column. These changes cannot be synchronized and require resetting the app server.
+To update any app level or tabel level files, or to modify the database schema, you will need to reset the app server. Make the changes on your PC as normal, push them to the device, and reset the app server.
 
     .. warning::
```

It would be useful to keep what 'modify the database schema' entails from the original description.